src/onegov/org/forms/page.py

```diff
@@ -61,6 +62,62 @@ class PageForm(PageBaseForm):
     )
 
 
+class IframeForm(PageBaseForm):
+    """ Defines the form for pages with the 'iframe' trait. """
+
+    allowed_domains: list[str] = []
+
+    domain_hint = PanelField(
+        text=_('There are currently no allowed domains for iframes. To enable '
+               'domains for iframes, please contact info@seantis.ch.'),
+        kind='callout',
+        fieldset=_('URL')
+    )
+
+    url = URLField(
+        label=_("URL"),
+        validators=[InputRequired(), URL(require_tld=False)],
+        fieldset=_('URL')
+    )
+
+    height = StringField(
+        label=_("Height"),
+        description=_("The height of the iframe in pixels. "
+                      "If not set, the iframe will have a standard height of "
+                      "800px."),
+        render_kw={'placeholder': 'auto'},
+        fieldset=_('Display')
+    )
+
+    as_card = BooleanField(
+        label=_("Display as card"),
+        description=_("Display the iframe as a card with a border"),
+        fieldset=_('Display')
+    )
+
+    def on_request(self) -> None:
+        self.allowed_domains = (
+            self.request.app.allowed_iframe_domains)  # type: ignore
+        if self.allowed_domains:
+            self.domain_hint.text = (
```

Comment:
There are hardcoded  domains such as youtube, vimeo etc.(https://github.com/OneGov/onegov-cloud/blob/master/src/onegov/org/app.py#L516). I guess it would make sense to mention these here as well.

You could even inspect the default content security policy here (`getattr(self.request.app.settings.content_security_policy.default, 'child_src', set())`).